lib/galaxy/tools/deps/resolvers/__init__.py

```diff
@@ -48,8 +48,15 @@ class MultipleDependencyResolver:
     """Variant of DependencyResolver that can optionally resolve multiple dependencies together."""
 
     @abstractmethod
-    def resolve_all( self, requirements, **kwds ):
-        """Given multiple requirements yield Dependency objects if and only if they may all be resolved together.
+    def resolve_all(self, requirements, **kwds):
+        """
+        Given multiple requirements yields a list of Dependency objects if and only if they may all be resolved together.
+
+        Unsuccessfull attempts should return an empty list.
+
+        :param requirements: list of tool requirements
+        :param type: [ToolRequirement] or ToolRequirements
```

Comment:
We are currently passing in ToolRequirements instances. The `requirements` attribute in `Tool` instances are instances of ToolRequirements objects ( https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/deps/requirements.py#L225). So we support both lists of `ToolRequirement` instances and `ToolRequirements` instances, but I think we are usually not using lists of ToolRequirement instances.